packages/manager/apps/pci-gateway/src/pages/list/ListGuard.tsx

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAllAggregatedGateway } from '@/api/hooks/useGateway';
+
+export default function ListGuard({
+  projectId,
+  children,
+}: {
+  projectId: string;
+  children: JSX.Element;
+}): JSX.Element {
+  const [displayChildren, setDisplayChildren] = useState(false);
+
+  const navigate = useNavigate();
+
+  const { data: aggregatedGateways, isPending } = useAllAggregatedGateway(
+    projectId,
+  );
+
+  useEffect(() => {
+    if (!isPending) {
+      if (aggregatedGateways?.length > 0) {
+        setDisplayChildren(true);
+      } else {
+        navigate(`/pci/projects/${projectId}/gateway/onboarding`);
+      }
+    }
+  }, [isPending, aggregatedGateways, navigate]);
+
+  return displayChildren ? children : null;
+}
```

Comment:
Instead of duplicating the component twice (ListGuard, OnboardingGuard), you could have only 1 component that takes the url to navigate to as param

Reply:
We need to have two components because the guard of onboarding forward on the listing page if we have data but the guard of listing page doesn't forward but display children we have data

In all cases, i have done the same kind of suggestion in this page that onboarding page :)